### PR TITLE
Knowledgebase images gallery

### PR DESCRIPTION
**Pull request by hsorby**
# Description

Upgrading the images gallery tab in the dataset details page.

## Type of change

Delete those that don't apply.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

## Common images

Using dataset 9: Influence of left vagal stimulus pulse parameters on vagal and gastric activity in rat
Relative URL: datasets/9?type=dataset&tab=images
On the images tab should see:

image

## Segmented data

Using dataset 43 (and dataset 37)
Relative URL: datasets/37?type=dataset&tab=images
Relative URL: datasets/43?type=dataset&tab=images
For dataset 43 the last images in the gallery should be:

image

 * dataset 37 annotation is incorrect and doesn't work.

## Scaffold data

Using dataset 99
Relative URL: datasets/99?type=dataset&tab=images
On the images tab should see:

image

## Video data

Using dataset 73
Relative URL: datasets/73?type=dataset&tab=images
On the images tab for the 16th index should see:

image

## Plot data

Using dataset 141
Relative URL: 141?type=dataset&tab=images&path=files
On the images tab should see:

image

 * due to annotation errors only 'subject1.txt' plot will be correctly displayed when clicking the view plot button.
 
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: nuxt.config.js [hunk 1]
```diff
@@ -29,11 +29,11 @@ export default {
   env: {
     portal_api: process.env.PORTAL_API_HOST || 'http://localhost:8000',
     flatmap_api:
-      process.env.FLATMAP_API_HOST || 'https://mapcore-demo.org/flatmaps/',
+      process.env.FLATMAP_API_HOST || 'https://mapcore-demo.org/devel/flatmap/v1/',
```
Comment (alan-wu) [inline nuxt.config.js:32]: Can we revert this line please?

File: package.json [hunk 2]
```diff
@@ -17,8 +17,10 @@
     "test:debug": "node --inspect node_modules/.bin/jest --runInBand"
   },
   "dependencies": {
-    "@abi-software/mapintegratedvuer": "0.1.31-fix-1",
-    "@abi-software/plotvuer": "^0.2.42",
+    "@abi-software/flatmapvuer": "0.1.36-beta-features-7",
```
Comment (alan-wu) [inline package.json:20]: Can we have this change to "@abi-software/flatmapvuer": "0.1.33" please?

File: mixins/request/index.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import EventBus from '@/static/js/event-bus'
+// import EventBus from '@/static/js/event-bus'
```
Comment (alan-wu) [inline mixins/request/index.js:1]: Revert changes in this file?

**Review by alan-wu — COMMENTED** [review]
Looks good to me, some minor changes to make sure we dont expose beta viewer and services. Thanks.

**Review by alan-wu — APPROVED** [review]
Looks good to me.

**Review by Tehsurfer — APPROVED** [review]
I've been using this branch for the last week and it seems to be working really well! Thanks @hsorby 

I haven't checked all the changes, but the changes in the image gallery viewer are much better than before. Much more data showing up than before, UI is more refined, and it is also now easier to work on.

I'd like to merge it in and continue forward on this branch

File: components/BiolucidaViewer/BiolucidaViewer.vue [hunk 4]
```diff
@@ -43,6 +47,24 @@ export default {
...
       }
     }
+  },
+  watch: {
+    queryView() {
+      this.$refs.biolucida.contentWindow.postMessage('getImgPos')
+    }
+  },
+  mounted() {
+    window.addEventListener('message', this.receiveMessage)
```
Comment (Tehsurfer) [inline components/BiolucidaViewer/BiolucidaViewer.vue:57]: I'm guessing Biolucida's image viewer only communicates through window message events?

File: components/BiolucidaViewer/BiolucidaViewer.vue [hunk 5]
```diff
@@ -43,6 +47,24 @@ export default {
...
+  },
+  beforeDestroy() {
+    window.removeEventListener('message', this.receiveMessage)
+  },
+  methods: {
+    receiveMessage(event) {
+      console.log('received message.')
```
Comment (Tehsurfer) [inline components/BiolucidaViewer/BiolucidaViewer.vue:64]: Probably don't need these anymore

File: components/ImagesGallery/ImagesGallery.vue [hunk 6]
```diff
@@ -1,120 +1,90 @@
...
-  <div class="biolucida-image-gallery full-size">
-    <div class="description-info">
+  <div class="full-size">
+    <div v-if="hasDescription" class="description-info">
       <p>
         <strong>Data collection:</strong>
         {{ description }}
       </p>
-      <p>
+      <!-- <p>
```
Comment (Tehsurfer) [inline components/ImagesGallery/ImagesGallery.vue:8]: This still needed?

File: components/BiolucidaViewer/BiolucidaViewer.vue (hunk 4 above)
Comment (hsorby) [reply components/BiolucidaViewer/BiolucidaViewer.vue:57]: This is for the postMessage messaging between the portal and Biolucida.

**Pull request merged**